Assets/Scripts/RoadControl.cs

```diff
@@ -355,125 +591,467 @@ void Update()
+                                                                            nearEdgeHeight + (droidHeight + bd.droidSprite.sprite.rect.height) * nearEdgeWidthScale / 2, -1f);
+                                bd.droidSprite.enabled = true;
+                            }
+
+                            //if ((Time.time - bd.dropDelayRefTime > delayBeforeDropping) && bd.droidSprite.sprite == droidHolding) {
+                            if ((bd.segmentIndex < curSegmentIndex + 200) && bd.droidSprite.sprite == droidHolding) {
+                                bd.droidSprite.sprite = droidEmptyHanded;
+                                Bomb bmb = new Bomb();
+                                GameObject obj = new GameObject();
+
+                                GameObject imgObj = new GameObject();
+                                RawImage img = imgObj.AddComponent<RawImage>();
+                                imgObj.transform.SetParent(warningRoadLinesParent);
+                                bmb.blip = img;
+                                bmb.blip.texture = warningRoadBombBlipSprite;
+                                bmb.blip.rectTransform.anchoredPosition = warningRoadLines[0].rectTransform.anchoredPosition;
+                                bmb.blip.rectTransform.sizeDelta = Vector2.zero;
+                                bmb.blip.enabled = true;
+
+                                GameObject imgProjObj = new GameObject();
+                                RawImage imgProj = imgProjObj.AddComponent<RawImage>();
+                                imgProjObj.transform.SetParent(warningRoadLinesParent);
+                                bmb.blipProjection = imgProj;
+                                bmb.blipProjection.texture = warningRoadBombBlipSprite;
+                                bmb.blipProjection.rectTransform.anchoredPosition = warningRoadLines[0].rectTransform.anchoredPosition +
+                                                                                            new Vector2(0.0f, bombBlipProjectionRoadOffsetY);
+                                bmb.blipProjection.rectTransform.sizeDelta = Vector2.zero;
+                                bmb.blipProjection.enabled = true;
+                                bmb.blipProjectionFlashRefTime = Time.time;
+
+                                obj.AddComponent<SpriteRenderer>().enabled = false;
+                                bmb.segmentIndex = bd.segmentIndex;
+                                bmb.singleBomb = obj.transform;
+                                bmb.offsetX = bd.offsetX;
+                                bmb.isExploding = false;
+                                bmb.explosionHeightOffset = 50f;
+                                bmb.bombSprite = obj.GetComponent<SpriteRenderer>();
+                                bmb.bombSprite.sprite = bombCamo;
+                                bmb.bombSprite.enabled = true;
+
+                                activeBombPool.Add(bmb);
+                            }
```

Comment:
There is a memory leak here from not tracking `GameObject`s created for images and their projections.

If you refactor this so that `Bomb` inherits from `MonoBehaviour`, you can make it easier to track image objects held by bombs (rather than just their image components.

```
if ((bd.segmentIndex < curSegmentIndex + 200) && bd.droidSprite.sprite == droidHolding) {
    bd.droidSprite.sprite = droidEmptyHanded;
    GameObject obj = new GameObject("Bomb");
    Bomb bmb = obj.AddComponent<Bomb>();

    bmb.InitializeBlip(warningRoadLinesParent,
        warningRoadLines[0].rectTransform.anchoredPosition, bombBlipProjectionRoadOffsetY, warningRoadBombBlipSprite);

    bmb.InitializeBomb(bd, bombCamo);

    activeBombPool.Add(bmb);
}
```

And here is the modified Bomb class I used for this example:
```
public class Bomb : MonoBehaviour
{
    public Transform singleBomb;
    public RawImage blip;
    public RawImage blipProjection;
    public SpriteRenderer bombSprite;
    public int segmentIndex;
    public float offsetX;
    public float explosionHeightOffset = 50f;
    public bool isExploding = false;
    public int explosionFrameIndex;
    public float explosionRefTime;
    public float blipProjectionFlashRefTime;

    GameObject imageObject;
    GameObject projectionObject;

    //runs before Start()
	private void Awake() {
        imageObject = new GameObject("image");
        projectionObject = new GameObject("projection");
        imageObject.AddComponent<SpriteRenderer>();
        projectionObject.AddComponent<SpriteRenderer>();

        blip = imageObject.AddComponent<RawImage>();
        blip.enabled = false;
        blip.rectTransform.sizeDelta = Vector2.zero;

        blipProjection = projectionObject.AddComponent<RawImage>();
        blipProjection.enabled = false;
        blipProjection.rectTransform.sizeDelta = Vector2.zero;

        blipProjectionFlashRefTime = Time.time;

        bombSprite = gameObject.AddComponent<SpriteRenderer>();
        bombSprite.enabled = false;

        singleBomb = transform;
	}

    public void InitializeBomb(BombDroid bombDroid, Sprite sprite) {
        segmentIndex = bombDroid.segmentIndex;
        offsetX = bombDroid.offsetX;
        bombSprite.sprite = sprite;
        bombSprite.enabled = true;
    }

    public void InitializeBlip(Transform parent, Vector2 anchoredPosition, float yOffset, Texture spriteTexture) {
        imageObject.transform.SetParent(parent);
        projectionObject.transform.SetParent(parent);
        blip.rectTransform.anchoredPosition = anchoredPosition;
        anchoredPosition.y += yOffset;
        blipProjection.rectTransform.anchoredPosition = anchoredPosition;
        blip.texture = spriteTexture;
        blipProjection.texture = spriteTexture;
        blip.enabled = true;
        blipProjection.enabled = true;
    }
}
```
```